setup.py

```diff
@@ -163,7 +163,7 @@ def get_files(directory, install_path, valid_suffixes):
         package_dir={"": "src"},
         install_requires=['pyparsing', 'fparser>=0.2.0', 'configparser',
                           'jsonschema', 'sympy', "Jinja2", 'termcolor',
-                          'graphviz'],
+                          'graphviz', 'sphinx'],
```

Comment:
I am not too keen on making 'shpinx' a required dependency, it is a huge dependency, see spack below.

It seems we use it to "pretty-print" types? Could we modify the "from sphinx.util.typing import stringify_annotation" so that if sphinx is not found we define a stringify_annotation method with: `retrun str(v.annotation)` or something simple. Development environments and when we generate the documentation will have the shpinx libs and will generate the nicer text? or do we use it for something else?

![image](https://github.com/user-attachments/assets/b6ca8b99-f869-49b8-b511-9e8857a2916a)

Reply:
From what I remember I had issues with 3.7 and 3.8 without the sphinx import and it not being able to turn it into a string, but I can take another look.

Reply:
Just pushed a change without sphinx being a required dependency, ended up on pragma no cover for the lines importing a local version of stringify_annotation when i couldn't import one from sphinx.